### PR TITLE
Validation bugs

### PR DESCRIPTION
**Pull request by RevylaV**
## Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

#### Screenshots

Original             |  Updated
:-------------------------:|:-------------------------:
** original screenshot **  |  ** updated screenshot **

### Checklist
- [x] 🔽 My branch is up-to-date with "development" branch
- [x] ✅All tests passed locally
- [x] ✨My changes working with up-to-date front-end and admin part locally, like charm

### CODE REVIEW

**Review by dieie32 — APPROVED** [review]

File: modules/constructor/constructor-front-pocket/constructor-front-pocket.model.js [hunk 1]
```diff
@@ -1,38 +1,28 @@
...
-const {
-  DB_COLLECTIONS_NAMES: { CONSTRUCTOR_FRONT_POCKET, MATERIAL, COLOR },
-} = require('../../../consts/db-collections-names');
 
 const constructorFrontPocketSchema = new mongoose.Schema({
   name: [Language],
   material: {
     type: mongoose.Schema.Types.ObjectId,
-    ref: MATERIAL,
+    ref: 'Material',
```
Comment (Haytham1337) [inline modules/constructor/constructor-front-pocket/constructor-front-pocket.model.js:13]: dont use magic strings please

File: modules/constructor/constructor-front-pocket/constructor-front-pocket.model.js [hunk 2]
```diff
@@ -1,38 +1,28 @@
...
   },
+  image: Yup.string().required(IMAGE_NOT_PROVIDED),
   basePrice: [CurrencySet],
   available: Boolean,
   default: Boolean,
 });
 
 module.exports = mongoose.model(
-  CONSTRUCTOR_FRONT_POCKET,
+  'ConstructorFrontPocket',
```
Comment (Haytham1337) [inline modules/constructor/constructor-front-pocket/constructor-front-pocket.model.js:26]: same

File: modules/constructor/constructor-basic/constructor-basic.model.js [hunk 3]
```diff
@@ -1,4 +1,5 @@
 const mongoose = require('mongoose');
+const Yup = require('yup');
```
Comment (Haytham1337) [inline modules/constructor/constructor-basic/constructor-basic.model.js:2]: we dont use yap validations on backend

**Comment by Kiyooooko** [comment]
pls resolve conflicts
Screenshot 2021-03-22 at 12 06 25

File: package.json [hunk 4]
```diff
@@ -84,7 +84,8 @@
     "start-server-and-test": "^1.11.3",
     "uniqid": "^5.2.0",
     "util": "^0.12.3",
-    "winston": "^3.3.3"
+    "winston": "^3.3.3",
+    "yup": "^0.32.9"
```
Comment (Kiyooooko) [inline package.json:88]: remove yup from json

File: modules/constructor/constructor-bottom/constructor-bottom.permissions.js [hunk 5]
```diff
@@ -2,15 +2,26 @@ const { allow } = require('graphql-shield');
...
   getAllConstructorBottom: allow,
   getConstructorBottomById: allow,
 };
 
 const сonstructorBottomPermissionsMutations = {
-  addConstructorBottom: hasRoles([ADMIN, SUPERADMIN]),
-  updateConstructorBottom: hasRoles([ADMIN, SUPERADMIN]),
+  addConstructorBottom:
+    (inputDataValidation(CONSTRUCTOR_ELEMENT, constructorValidator),
+    hasRoles([ADMIN, SUPERADMIN])),
```
Comment (Kiyooooko) [inline modules/constructor/constructor-bottom/constructor-bottom.permissions.js:21]: use 'and' from graphql-shield to unite two middleware

File: modules/constructor/constructor-bottom/constructor-bottom.permissions.js [hunk 6]
```diff
@@ -2,15 +2,26 @@ const { allow } = require('graphql-shield');
...
+  updateConstructorBottom:
+    (inputDataValidation(CONSTRUCTOR_ELEMENT, constructorValidator),
+    hasRoles([ADMIN, SUPERADMIN])),
```
Comment (Kiyooooko) [inline modules/constructor/constructor-bottom/constructor-bottom.permissions.js:24]: same

File: modules/constructor/constructor-front-pocket/constructor-front-pocket.permission.js [hunk 7]
```diff
@@ -2,15 +2,26 @@ const { allow } = require('graphql-shield');
...
   getAllConstructorFrontPocket: allow,
   getConstructorFrontPocketById: allow,
 };
 
 const constructorFrontPocketPermissionsMutations = {
-  addConstructorFrontPocket: hasRoles([ADMIN, SUPERADMIN]),
-  updateConstructorFrontPocket: hasRoles([ADMIN, SUPERADMIN]),
+  addConstructorFrontPocket:
+    (inputDataValidation(CONSTRUCTOR_ELEMENT, constructorValidator),
+    hasRoles([ADMIN, SUPERADMIN])),
```
Comment (Kiyooooko) [inline modules/constructor/constructor-front-pocket/constructor-front-pocket.permission.js:21]: same

File: modules/constructor/constructor-front-pocket/constructor-front-pocket.permission.js [hunk 8]
```diff
@@ -2,15 +2,26 @@ const { allow } = require('graphql-shield');
...
+  updateConstructorFrontPocket:
+    (inputDataValidation(CONSTRUCTOR_ELEMENT, constructorValidator),
+    hasRoles([ADMIN, SUPERADMIN])),
```
Comment (Kiyooooko) [inline modules/constructor/constructor-front-pocket/constructor-front-pocket.permission.js:24]: same

File: validators/constructor.validator.js [hunk 9]
```diff
@@ -0,0 +1,25 @@
...
+const { userNameRegExp } = require('../consts/regexp');
+
+const constructorValidator = Joi.object({
+  name: Joi.array().items({
+    lang: Joi.string()
+      .trim()
+      .required(),
+    value: Joi.string()
+      .trim()
+      .regex(userNameRegExp),
```
Comment (Kiyooooko) [inline validators/constructor.validator.js:12]: value is .required() too

File: modules/constructor/constructor-basic/constructor-basic.permission.js [hunk 10]
```diff
@@ -2,15 +2,26 @@ const { allow } = require('graphql-shield');
...
   getAllConstructorBasics: allow,
   getConstructorBasicById: allow,
 };
 
 const constructorBasicPermissionsMutations = {
-  addConstructorBasic: hasRoles([ADMIN, SUPERADMIN]),
-  updateConstructorBasic: hasRoles([ADMIN, SUPERADMIN]),
+  addConstructorBasic:
+    (inputDataValidation(CONSTRUCTOR_ELEMENT, constructorValidator),
+    hasRoles([ADMIN, SUPERADMIN])),
```
Comment (Kiyooooko) [inline modules/constructor/constructor-basic/constructor-basic.permission.js:21]: use 'and' from graphql-shield to unite two middleware

File: modules/constructor/constructor-basic/constructor-basic.permission.js [hunk 11]
```diff
@@ -2,15 +2,26 @@ const { allow } = require('graphql-shield');
...
+  updateConstructorBasic:
+    (inputDataValidation(CONSTRUCTOR_ELEMENT, constructorValidator),
+    hasRoles([ADMIN, SUPERADMIN])),
```
Comment (Kiyooooko) [inline modules/constructor/constructor-basic/constructor-basic.permission.js:24]: use 'and' from graphql-shield to unite two middleware

File: validators/constructor.validator.js [hunk 12]
```diff
@@ -0,0 +1,25 @@
...
+  color: Joi.object().required(),
+  image: Joi.string().required(),
+  basePrice: Joi.array().items(
+    { currency: Joi.string() },
+    { value: Joi.string() }
+  ),
+});
+
+module.exports = {
+  constructorValidator,
```
Comment (Kiyooooko) [inline validators/constructor.validator.js:24]: your validation schema does not meet with graphQL input
Screenshot 2021-03-23 at 18 01 42
Screenshot 2021-03-23 at 18 03 40
Screenshot 2021-03-23 at 18 03 53

**Comment by AndriyLymych** [comment]
image
fix duplications

**Comment by AndriyLymych** [comment]
image
link pr

File: validators/constructor-bottom.validator.js [hunk 13]
```diff
@@ -0,0 +1,44 @@
...
+  }),
+  material: Joi.object().required(),
+  color: Joi.object().required(),
+  image: Joi.string().required(),
+  basePrice: Joi.array().items(
+    { currency: Joi.string() },
+    { value: Joi.string() }
+  ),
+});
+
```
Comment (AndriyLymych) [inline validators/constructor-bottom.validator.js:40]: its the same func that`addConstructorBottomValidator `

File: validators/constructor-bottom.validator.js [hunk 14]
```diff
@@ -0,0 +1,44 @@
...
+    value: Joi.string()
+      .trim()
+      .regex(userNameRegExp),
+  }),
+  material: Joi.object().required(),
+  color: Joi.object().required(),
+  image: Joi.string().required(),
+  basePrice: Joi.array().items(
+    { currency: Joi.string() },
+    { value: Joi.string() }
```
Comment (AndriyLymych) [inline validators/constructor-bottom.validator.js:19]: incorrect record

**Pull request closed without merging**